### PR TITLE
Stats routetesting

### PR DESCRIPTION
**Pull request by mattjackmcgregor**
fixed a whole shibang of conflicts! 
Tests passing and yarn dev running website sweet

### CODE REVIEW

File: client/actions/index.js [hunk 1]
```diff
@@ -1,7 +1,12 @@
+import {request} from 'superagent'
+
 export const SHOW_ERROR = 'SHOW_ERROR'
 export const CLEAR_ERROR = 'CLEAR_ERROR'
 export const SHOW_SUCCESS = 'SHOW_SUCCESS'
 
+export const REQUEST_COMPANYINFO = 'REQUEST_COMPANYINFO'
+export const RECEIVE_COMPANYINFO = 'RECEIVE_COMPANYINFO'
```
Comment (don-smith) [inline client/actions/index.js]: These action types are already defined in `actions/companies.js`. We should remove these and ensure that any file that needs these constants are getting them from `companies.js`.

File: tests/server/routes/statRoute.test.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+const request = require('supertest')
+const server = require('../../../server/server')
+
+jest.mock('../../../server/db/stats', () => ({
+  categoryMaleFemaleCount: () => Promise.resolve({
+    gender:
+  [ {Male: 3, Female: 10, Category: 'creative'},
+    {Male: 10, Female: 9, Category: 'support'},
+    {Male: 7, Female: 11, Category: 'technical'} ]
```
Comment (don-smith) [inline tests/server/routes/statRoute.test.js]: change formatting to:
```
    gender: [
      {Male: 3, Female: 10, Category: 'creative'},
      {Male: 10, Female: 9, Category: 'support'},
      {Male: 7, Female: 11, Category: 'technical'}
    ]
```

File: tests/server/routes/statRoute.test.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  }),
+  catetoryLocalForeignCount: () => Promise.resolve({
+    local: [{name: 'Local', value: 69}, {name: 'Foreign', value: 31}]
```
Comment (don-smith) [inline tests/server/routes/statRoute.test.js]: change formatting to:
```
local: [
  {name: 'Local', value: 69},
  {name: 'Foreign', value: 31}
]
```

File: tests/server/routes/statRoute.test.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  expect(true).toBeTruthy()
+})
+
+test('GET /api/v1/stats', () => {
+  return request(server)
+    .get('/api/v1/stats')
+    .expect('Content-Type', /json/)
+    .expect(200)
+    .then(res => {
+      expect(res.body).toHaveProperty('gender')
```
Comment (don-smith) [inline tests/server/routes/statRoute.test.js]: are there other properties that would be good to assert?

File: tests/server/routes/statRoute.test.js (hunk 2 above)
Comment (mattjackmcgregor) [reply tests/server/routes/statRoute.test.js]: done

File: tests/server/routes/statRoute.test.js (hunk 3 above)
Comment (mattjackmcgregor) [reply tests/server/routes/statRoute.test.js]: done

**Pull request merged**